packages/flutter/lib/src/widgets/semantics_debugger.dart

```diff
@@ -332,14 +332,14 @@ class _SemanticsDebuggerPainter extends CustomPainter {
     return childrenDepth + 1;
   }
 
-  void _paint(Canvas canvas, SemanticsNode node, int rank) {
+  void _paint(Canvas canvas, SemanticsNode node, int rank, int indexInParent) {
     canvas.save();
     if (node.transform != null) {
       canvas.transform(node.transform!.storage);
     }
     final Rect rect = node.rect;
     if (!rect.isEmpty) {
-      final Color lineColor = Color(0xFF000000 + math.Random(node.id).nextInt(0xFFFFFF));
+      final Color lineColor = Color(0xFF000000 + math.Random(_getColorSeed(rank, indexInParent)).nextInt(0xFFFFFF));
```

Comment:
question: so it looks like `_findDepth` finds the distance between this node and the deepest descendant. Why this instead of using the distance to the root? If we use that depth instead, all root nodes will be of the same color and every node at the same "location" in the tree (`(# of ancestors, child index)` pair) will have a consistent color? Is that worth doing?